appgate/client.py

```diff
@@ -115,28 +114,67 @@ async def init(self) -> None:
             namespace=self.namespace
         )
         self._configmap_mt = configmap.metadata
-        self._entries = {
-            k: load_latest_entity_generation(k, v) for k, v in (configmap.data or {}).items()
-        }
+        self._data = configmap.data or {}
+
+    @staticmethod
+    def _entry_key(key: str) -> str:
+        return f'entry.{key}'
+
+    @staticmethod
+    def _device_id_key() -> str:
+        return 'device-id'
+
+    async def ensure_device_id(self) -> str:
+        """
+        Try to get the device id from the config map.
+        If that fails, generate one and store it in the configmap.
+        """
+        try:
+            device_id = self._data[self._device_id_key()]
+        except KeyError:
+            device_id = str(uuid.uuid4())
+            self._data[self._device_id_key()] = device_id
```

Comment:
This will generate a new uuid everytime is restarted right? Unless it's defined in the configmap.

Reply:
Yeah that's the point, it takes it from the config map if its there, else it generates and stores it.
I could return early when its found though.